subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -73,6 +73,10 @@ public JavaCompile() {
         CompileOptions compileOptions = getServices().get(ObjectFactory.class).newInstance(CompileOptions.class);
         this.compileOptions = compileOptions;
         CompilerForkUtils.doNotCacheIfForkingViaExecutable(compileOptions, getOutputs());
+
+        // this mimics the behavior of the Ant javac task (and therefore AntJavaCompiler),
+        // which silently excludes files not ending in .java
+        include("**/*.java");
```

Comment:
Why do we need this now?

Reply:
I removed the filtering from `NormalizingJavaCompiler` - so we only do it here.

Reply:
By using `include("**/*.java")`, does that also mean we now ignore empty directories in the source root?